### PR TITLE
fix: dont crash if resServer doesn't exist

### PR DESCRIPTION
**Pull request by aronwk-aaron**
and try to make resServer as well if it doesn't exist

Tested that this works

### CODE REVIEW

File: dMasterServer/MasterServer.cpp [hunk 1]
```diff
@@ -154,6 +154,15 @@ int main(int argc, char** argv) {
 	const bool cdServerExists = std::filesystem::exists(BinaryPathFinder::GetBinaryDir() / "resServer" / "CDServer.sqlite");
 	const bool oldCDServerExists = std::filesystem::exists(Game::assetManager->GetResPath() / "CDServer.sqlite");
 	const bool fdbExists = std::filesystem::exists(Game::assetManager->GetResPath() / "cdclient.fdb");
+	const bool resServerPathExists = std::filesystem::is_directory(BinaryPathFinder::GetBinaryDir() / "resServer");
+
+	if (!resServerPathExists) {
+		LOG("%s does not exist, creating it.", (BinaryPathFinder::GetBinaryDir() / "resServer").c_str());
```
Comment (EmosewaMC) [inline dMasterServer/MasterServer.cpp:160]: Lets do this concatenation of binary dir and resServer once
also .c_str on a std::filesystem::path returns a wstring on windows, so this logger call should be .string().c_str()

Comment (aronwk-aaron) [reply dMasterServer/MasterServer.cpp:160]: was following the convention below with the .c_str()

Comment (EmosewaMC) [reply dMasterServer/MasterServer.cpp:160]: yes and that is incorrect and should be changed as well.  Unix uses std::string for its file paths and windows uses std::wstring for its

**Review by Xiphoseer — APPROVED** [review]

**Review by EmosewaMC — COMMENTED** [review]
there is a bunch of extra copies being done with all the .string() calls but its not enough to block.  lgtm

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**